src/ucp/core/ucp_request.h

```diff
@@ -149,17 +151,12 @@ struct ucp_request {
                 struct {
                     uint64_t               message_id;  /* used to identify matching parts
                                                            of a large message */
-                    union {
-                        ucs_ptr_map_key_t  sreq_id;     /* send request ID on the
-                                                           sender side */
-                        ucs_ptr_map_key_t  rreq_id;     /* receive request ID on the
-                                                           recv side (used in AM rndv) */
-                    };
+                    /* Remote request ID received from a peer (used in AM rndv) */
+                    ucs_ptr_map_key_t      remote_req_id;
 
                     union {
-                        struct {
-                            ucp_tag_t         tag;
-                            ucs_ptr_map_key_t req_id;
+                        union {
```

Comment:
why need a union with single element? Can we add `remote_req_id` here?

Reply:
> why need a union with single element?

it is too many changes to fix `tag.tag` - better to do it in a separate PR

> Can we add remote_req_id here?

we don't need `remote_req_id`

Reply:
i meant can we move it from line 155?

Reply:
> it is too many changes to fix `tag.tag` - better to do it in a separate PR

why not here? It is related refactoring

Reply:
> i meant can we move it from line 155?

I see - will try to move

Reply:
also, need to remove `tag` name from union, since the `remote_req_id` can be used for AM and TAG